modules/oe_bootstrap_theme_helper/tests/src/Kernel/BclComponentsOverrideTest.php

```diff
@@ -45,34 +41,39 @@ protected function setUp(): void {
       'build',
     ];
     new Settings($settings);
-
-    // Create testing BCL components and patterns.
-    $this->createTestingPatterns();
   }
 
   /**
    * Tests that BCL components are overridable in sub-themes.
    *
    * @param string $theme
-   *   The theme being installed.
-   * @param array $patterns
-   *   Associative array of patterns to be tested in $theme, having the pattern
-   *   ID as keys and the expected pattern output as values.
+   *   The theme to set as default.
+   * @param array $templates
+   *   Associative array of templates to be tested in $theme, having the
+   *   template name as keys and the expected path parts as values.
+   *   The path parts consist of the theme name and the relative path of the
+   *   expected template to be loaded inside the theme folder.
    *
    * @dataProvider bclComponentsOverridingTestCasesProvider
    */
-  public function testBclComponentsOverriding(string $theme, array $patterns): void {
+  public function testBclComponentsOverriding(string $theme, array $templates): void {
     // Install the theme. Subsequently, this installs also the base themes.
     $this->container->get('theme_installer')->install([$theme]);
     $this->config('system.theme')->set('default', $theme)->save();
+    $twig_environment = $this->container->get('twig');
+
+    foreach ($templates as $template_name => $path_parts) {
+      // Generate the path to the expected theme and template.
+      $expected_path = drupal_get_path('theme', $path_parts[0]) . $path_parts[1];
```

Comment:
```suggestion
      $expected_path = $theme_list_service->getPath($path_parts[0]) . $path_parts[1];
```
and add $theme_list_service = $this->container->get('extension.list.theme'); before the loop